ccm_web/client/src/components/MultipleUserEnrollmentWorkflow.tsx

```diff
@@ -0,0 +1,357 @@
+    const requirements = (
+      <>
+      <Typography>
+        The file includes four columns of data with corresponding headers:
+      </Typography>
+      <ul>
+        <li>
+          <Typography>
+            &quot;{EMAIL_HEADER.toLowerCase()}&quot; with each user&apos;s external email address (not from University of Michigan);
+          </Typography>
+        </li>
+        <li>
+          <Typography>
+            &quot;{ROLE_HEADER.toLowerCase()}&quot; with one of the Canvas roles you are allowed to enroll users as (see documentation);
+          </Typography>
+        </li>
+        <li>
+          <Typography>
+          &quot;{FIRST_NAME_HEADER.toLowerCase()}&quot; with the user&apos;s first name (between one and 255 characters in length);
+          </Typography>
+        </li>
+        <li>
+          <Typography>
+            and &quot;{LAST_NAME_HEADER.toLowerCase()}&quot; with the user&apos;s last name (between one and 255 characters in length).
+          </Typography>
+        </li>
+      </ul>
+      </>
+    )
+    const fileData = (
+      `${EMAIL_HEADER},${ROLE_HEADER},${FIRST_NAME_HEADER},${LAST_NAME_HEADER}\n` +
+      'jdoe@example.edu,student,jane,doe'
+    )
+
+    const handleBackClick = (): void => {
+      handleResetUpload()
+      setActiveStep(CSVWorkflowStep.Select)
+    }
+
+    const handleValidation = (headers: string[] | undefined, rowData: CSVRecord[]): void => {
+      const schemaValidator = new CSVSchemaValidator<ExternalEnrollmentRecord>(
+        REQUIRED_HEADERS, isExternalEnrollmentRecord, 200
+      )
+      const validationResult = schemaValidator.validate(headers, rowData)
+      if (!validationResult.valid) return setSchemaInvalidations(validationResult.schemaInvalidations)
+
+      const externalRecords = validationResult.validData
+      const errors: EnrollmentInvalidation[] = []
+
+      const emails = externalRecords.map(r => r[EMAIL_HEADER])
+      const emailValidators = [new EmailRowsValidator(), new DuplicateEmailRowsValidator()]
+      emailValidators.map(validator => errors.push(...validator.validate(emails)))
+
+      const firstNames = externalRecords.map(r => r[FIRST_NAME_HEADER])
+      const firstNameValidator = new FirstNameRowsValidator()
+      errors.push(...firstNameValidator.validate(firstNames))
+
+      const lastNames = externalRecords.map(r => r[LAST_NAME_HEADER])
+      const lastNameValidator = new LastNameRowsValidator()
+      errors.push(...lastNameValidator.validate(lastNames))
+
+      const roles = externalRecords.map(r => r[ROLE_HEADER])
+      const rolesValidator = new RoleRowsValidator()
+      errors.push(...rolesValidator.validate(roles, props.rolesUserCanEnroll))
+
+      if (errors.length > 0) return setRowInvalidations(errors)
+
+      const externalEnrollments: AddNumberedNewExternalUserEnrollment[] = externalRecords.map((r, i) => ({
+        rowNumber: i + 2,
+        email: r[EMAIL_HEADER],
+        role: r[ROLE_HEADER] as ClientEnrollmentType,
```

Comment:
The `RoleRowsValidator` uses `isValidRole`, which is also a type guard, but that type information isn't available here. I think based on the context and logic, it's safe to type cast, but there may be a better way to do it.